docs/reference/ebpf-traffic-collection.md

```diff
@@ -0,0 +1,112 @@
+
+eBPF programs run directly in kernel space, providing unprecedented visibility into network traffic with minimal overhead. Unlike traditional packet capture methods that require copying packets from kernel to user space, eBPF can process network data directly where it originates.
+
+Key advantages include:
+- **Zero-copy operations**: Data can be analyzed without expensive memory copies
+- **Minimal performance impact**: Programs run in kernel space with native performance
+- **Real-time processing**: Traffic analysis happens as packets flow through the network stack
+- **Comprehensive visibility**: Access to both network packets and kernel metadata
+
+### Traffic Capture Architecture
+
+eBPF programs attach to various kernel hook points to intercept network traffic:
+
+#### Socket-Level Hooks
+- **Socket create/destroy**: Track connection lifecycle
+- **Socket send/receive**: Monitor data transmission
+- **Socket state changes**: Observe connection state transitions
+
+#### Network Stack Hooks
+- **TC (Traffic Control)**: Capture packets at the network interface level
+- **XDP (eXpress Data Path)**: Ultra-fast packet processing at the driver level
+- **Netfilter hooks**: Integrate with iptables and connection tracking
+
+#### System Call Tracing
+- **syscall entry/exit**: Monitor application network API calls
+- **kprobes/kretprobes**: Trace specific kernel functions
+- **tracepoints**: Use predefined kernel instrumentation points
+
+### Data Collection Process
+
+1. **Program Loading**: eBPF programs are compiled to bytecode and loaded into the kernel
+2. **Verification**: Kernel verifier ensures programs are safe and will terminate
+3. **Attachment**: Programs attach to appropriate kernel hooks
+4. **Event Generation**: Network events trigger program execution
+5. **Data Aggregation**: Programs collect and process network metadata
+6. **User Space Delivery**: Processed data is delivered to user space applications
+
+### TCP Traffic Analysis
+
+For TCP traffic specifically, eBPF enables deep inspection capabilities:
+
+#### Connection Tracking
+- Monitor TCP handshake establishment
+- Track connection state throughout lifecycle
+- Detect connection failures and anomalies
+
+#### Data Flow Analysis
+- Capture request/response pairs
+- Measure latency and throughput
+- Analyze data patterns and sizes
+
+#### Performance Metrics
+- Round-trip time (RTT) measurements
+- Congestion window analysis
+- Retransmission detection
+- Bandwidth utilization
+
+### Security and Safety
+
+eBPF's design ensures safe execution within the kernel:
+
+- **Verifier**: Static analysis prevents infinite loops and invalid memory access
+- **Sandboxing**: Programs run in isolated execution context
+- **Resource limits**: Bounded execution time and memory usage
+- **Permission model**: Fine-grained access control to kernel features
+
+## Implementation Considerations
+
+### Kernel Requirements
+- Linux kernel 4.4+ (5.0+ recommended for advanced features)
```

Comment:
For our purposes we need 5.1+, which is the current LTS kernel version that RHEL uses